plugins/catalog-graphql/src/graphql/module.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Logger } from 'winston';
+import fs from 'fs';
+import path from 'path';
+import { GraphQLModule } from '@graphql-modules/core';
+import { Resolvers, CatalogEntity, CatalogQuery } from './types';
+import { Entity } from '@backstage/catalog-model';
+import { Config } from '@backstage/config';
+import { CatalogClient } from '../service/client';
+
+export interface ModuleOptions {
+  logger: Logger;
+  config: Config;
+}
+
+const parseToCatalogEntities = (e: Entity): CatalogEntity => ({
+  ...e,
+  metadata: {
+    ...e.metadata,
+    annotations: Object.entries(
+      e.metadata.annotations ?? {},
+    ).map(([key, value]) => ({ key, value })),
```

Comment:
oh right ... yeah if we want to type this dict we need to do something like this, but you could also declare it as type `JSON` or even `JSONObject` in the schema - that leaves it as an undefined json blob which is kinda true that it is. There may be some concerns then about what the future of custom scalars looks like in node graphql. But I think we should look into that.

https://www.apollographql.com/docs/apollo-server/schema/scalars-enums/

https://github.com/taion/graphql-type-json

then we should be able to return the entities verbatim without the parsing

Reply:
I have done this for annotations + will do it for the `schema` field which powers the form in the `Templates` entity rather than wrapping that in types.